server/src/main/java/org/opensearch/index/engine/InternalEngine.java

```diff
@@ -2336,26 +2338,27 @@ private IndexWriterConfig getIndexWriterConfig() {
         iwc.setMergeScheduler(mergeScheduler);
         // Give us the opportunity to upgrade old segments while performing
         // background merges
-        MergePolicy mergePolicy = config().getMergePolicy();
+        this.noMergeOnFlushPolicy = config().getMergePolicy();
```

Comment:
shall we make this change to the subclasses of Internal Engine like IngestionEngine?

Reply:
If we change `InternalEngine`, then `IngestionEngine` can directly use this change.